src/components/external-dns/index.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+  Fragment,
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useState,
+} from 'react';
+
+import * as PropTypes from 'prop-types';
+import { Checkbox, Icon, Table, Typography } from '@equinor/eds-core-react';
+import { ExternalDns, Tls } from '../../store/radix-api';
+import { ExternalDNSStatusBadge } from '../status-badges';
+import { chevron_down, chevron_up } from '@equinor/eds-icons';
+import clsx from 'clsx';
+import { TLSCertificateList } from '../tls-certificate-list';
+import { Alert, AlertProps } from '../alert';
+import { differenceInDays } from 'date-fns';
+import { pluraliser } from '../../utils/string';
+import { dataSorter, sortCompareString } from '../../utils/sort-utils';
+
+const dayPluraliser = pluraliser('day', 'days');
+
+function useGetSortedExternalDNS(
+  externalDnsList: Array<ExternalDns>
+): Array<ExternalDns> {
+  const [sortedData, setSortedData] = useState(externalDnsList);
+
+  useEffect(() => {
```

Comment:
consider useMemo instead of useEffect when required, but consider changing the data when you receive it (f.eks with `selectFromResult` in any UseQuery hook)

Reply:
replace function with useMemo